test/finch/http2/pool_test.exs

```diff
@@ -235,9 +235,6 @@ defmodule Finch.HTTP2.PoolTest do
       ])
 
       assert_receive {:resp, {:ok, _}}
-
-      assert_recv_frames([rst_stream(stream_id: ^stream_id, error_code: :no_error)])
-
```

Comment:
@sneako tbh I'm not sure what this test is/was testing, it's not asserting on:

```elixir
assert_receive {:resp, {:error, %Finch.Error{reason: :request_timeout}}}
```

like other similar tests do.

Reply:
@zachallaun blast from the past! Could you help changing this test? Or if it outlived its usefulness, we let it go?

Reply:
Nevermind, you showed up in git blame but I think you were just moving code around, sorry about that!

Reply:
I think its just testing the happy path? Seems fine to me to delete this assertion, let's do it

Reply:
@sneako per:

```elixir
test "request timeout with timeout > 0 that fires after request is done"
```

it makes sense we don't assert on any returned errors because there aren't any. Perhaps this test was asserting the side effects (a frame with no_error from Mint) but  since there aren't any, perhaps it's not so useful anymore. Probably OK to to keep it though!